local_vars/local_vars.cc

```diff
@@ -47,16 +47,29 @@ class LocalNameInserter {
 
     // Map through the reference structure, naming the locals, and preserving
     // the outer structure for the namer proper.
-    NamedArg nameArg(ast::ExpressionPtr arg) {
+    NamedArg nameArg(core::MutableContext ctx, const vector<NamedArg> &seen, ast::ExpressionPtr arg, uint32_t pos) {
         NamedArg named;
         auto *cursor = &arg;
 
+        auto mangleKeyword = [&ctx, &seen, pos](bool isKeyword, core::NameRef name) {
+            // Mangle any duplicate keyword arguments.
+            if (isKeyword) {
+                auto it = absl::c_find_if(seen, [name](auto &existing) { return existing.name == name; });
+                if (it != seen.end()) {
+                    return ctx.state.freshNameUnique(core::UniqueNameKind::MangledKeywordArg, name, pos);
+                }
+            }
+
+            return name;
+        };
```

Comment:
Should we make this a helper function, not a lambda? I don't think we should need to be allocating a closure every time this function gets called.